numpy/core/_internal.py

```diff
@@ -305,6 +305,106 @@ def _index_fields(ary, fields):
     copy_dtype = {'names':view_dtype['names'], 'formats':view_dtype['formats']}
     return array(view, dtype=copy_dtype, copy=True)
 
+# get the list of offsets of every field in a dtype
+def get_all_field_offests(dtype, base_offset=0):
+    fields = []
+    if dtype.fields is not None:
+        for name in dtype.names:
+           sub_dtype = dtype.fields[name][0]
+           sub_offset = dtype.fields[name][1] + base_offset
+           fields.extend(get_all_field_offests(sub_dtype, sub_offset))
+    else:
+        if dtype.shape:
+            sub_offsets = get_all_field_offests(dtype.base, base_offset)
+            count = 1
+            for dim in dtype.shape:
+                count *= dim
+            fields.extend((typ,off + dtype.base.itemsize*j)
+                                              for j in range(count)
+                                              for typ,off in sub_offsets)
+        else:
+            fields.append((dtype, base_offset))
+    return fields
+
+# perform object memory overlap tests (see _view_is_safe)
+def check_field_overlap(new_fields, old_fields):
+    from .numerictypes import object_
+    from .multiarray import dtype
+
+    #first go byte by byte and check we do not access bytes not in old_fields
```

Comment:
There are smarter ways of going about this check, but we probably shouldn't let micro-optimizations distract us from the larger goal.